signimage/signimage.go

```
@@ -0,0 +1,357 @@
+	extractiondir, err := os.MkdirTemp("/tmp/", "kmod_signer")
+	if err != nil {
+		fmt.Errorf("could not create temp dir: %v\n", err)
+		panic(err)
+	}
+
+	// sets up a tar archive we will use for a new layer
+	var b bytes.Buffer
+	tarwriter := tar.NewWriter(&b)
+
+	//make a map of the files to sign so we can track what we want to sign
+	kmodstosign := make(map[string]string)
+	for _, x := range strings.Split(fileslist, ":") {
+		kmodstosign[x] = "not found"
+	}
+
+	// set up image download otions
+	opts := make([]name.Option, 0)
+	ref, err := name.ParseReference(unsignedimagename, opts...)
+	if err != nil {
+		fmt.Errorf("could not parse the container image name: %v\n", err)
+		panic(err)
+	}
+
+	a, err := getAuthFromFile(pullsecret, strings.Split(unsignedimagename, "/")[0])
+	if err != nil {
+		fmt.Printf("failed to get auth: %v\n", err)
+		panic(err)
+	}
+
+	descriptor, err := remote.Get(ref, remote.WithAuth(a))
+	if err != nil {
+		te := &transport.Error{}
+
+		if errors.As(err, &te) && te.StatusCode == http.StatusNotFound {
+			fmt.Errorf("could not get image: http.StatusNotFound")
+		}
+
+		fmt.Errorf("could not get image: %v", err)
+		panic(err)
+	}
+
+	img, err := descriptor.Image()
+	if err != nil {
+		fmt.Errorf("could not Image(): %v", err)
+		panic(err)
+	}
+	fmt.Printf("\n== Successfully pulled image: %s\n", unsignedimagename)
+	fmt.Printf("\n== Looking for files: %s\n", strings.Replace(fileslist, ":", " ", -1))
+
+	/*
+	** loop through all the layers in the image from the top down
+	 */
+	layers, _ := img.Layers()
+	for i := len(layers) - 1; i >= 0; i-- {
+		fmt.Printf("== Searching layer %d\n", i)
+		currentlayer := layers[i]
+		layerreader, err := currentlayer.Uncompressed()
+		if err != nil {
+			fmt.Errorf("could not get layer: %v", err)
+			panic(err)
+		}
+
+		/*
+		** loop through all the files in the layer
+		 */
+		tarreader := tar.NewReader(layerreader)
+		for {
+			header, _ := tarreader.Next()
```

We should not ignore the error here.